Add jitpack support for Remixer.

miguelandres opened the pull request:
This was tested using the :jitpack-SNAPSHOT identifier so it tried against this very branch.

It works. Will be properly integrated in the next release.

Fixes #45.

Commit:
Add jitpack support for Remixer.

pingpongboss commented on README.md:
```diff
@@ -44,6 +44,48 @@ The project is defined as a gradle project with submodules.
+
+### If you're using the annotation-based API
+
+_(You most-likely want to do this)_. You need to add a couple more dependencies.
```
"most likely"

miguelandres replied:
Done

pingpongboss commented on README.md:
````diff
@@ -44,6 +44,48 @@ The project is defined as a gradle project with submodules.
+}
+```
+
+On the modules where you will be using Remixer `apply plugin: 'android-apt'`, and add an extra dependency on `provided 'com.github.material-foundation.material-remixer-android:remixer_annotation:develop-SNAPSHOT'`. Notice how the dependency is a `provided` clause instead of `compile`, this is on purpose as this is not a regular dependency but a compiler plugin.
````
I would give the build.gradle code block below as well

miguelandres replied:
Done.

Commit:
Add code block for annotation dependency example and fix extra hyphenation

Commit:
Merging the instructions to use annotation-based API into the main section.

Commit:
Add missing apply plugin clause.

Commit:
Improve wording on dependency instructions

Commit:
Moving the section on depending on Remixer above the Building section.

This makes more sense for developers. Also made "Installing the example app" a subheading of Building, since it assumes you've built it already.
